convert2rhel/actions/pre_ponr_changes/handle_packages.py

```diff
@@ -60,16 +80,40 @@ def run(self):
         logger.task("Convert: Remove excluded packages")
         logger.info("Searching for the following excluded packages:\n")
         try:
-            pkgs_to_remove = pkghandler._get_packages_to_remove(system_info.excluded_pkgs)
-            pkghandler.remove_pkgs_unless_from_redhat(pkgs_to_remove)
+            pkgs_to_remove = sorted(pkghandler._get_packages_to_remove(system_info.excluded_pkgs))
+            pkgs_removed = sorted(pkghandler.remove_pkgs_unless_from_redhat(pkgs_to_remove))
 
             # TODO: Handling SystemExit here as way to speedup exception
             # handling and not refactor contents of the underlying function.
         except SystemExit as e:
             # TODO(r0x0d): Places where we raise SystemExit and need to be
             # changed to something more specific.
             #   - When we can't remove a package.
-            self.set_result(level="ERROR", id="PACKAGE_REMOVAL_FAILED", message=str(e))
+            self.set_result(level="ERROR", id="EXCLUDED_PACKAGE_REMOVAL_FAILED", message=str(e))
+            return
+
+        # shows which packages were not removed, if false, all packages were removed
+        pkgs_not_removed = sorted(frozenset(pkgs_to_remove).difference(pkgs_removed))
+
+        if pkgs_not_removed:
+            pkg_names = _get_pkg_names(pkgs_not_removed)
+            message = "The following packages were not removed: %s" % ", ".join(pkg_names)
+            logger.warning(message)
+            self.add_message(
+                level="WARNING",
+                id="EXCLUDED_PACKAGES_NOT_REMOVED",
+                message=message,
+            )
+            return
+        if pkgs_removed:
+            message = "The following packages were removed: %s" % ", ".join(pkgs_removed)
+            logger.info(message)
+            self.add_message(
+                level="INFO",
+                id="EXCLUDED_PACKAGES_REMOVED",
+                message=message,
+            )
+            return
```

Comment:
I notice we have return in both the if statements. Is it possible that we remove some packages but not all, so that both should add a message?

Reply:
Before we had an if/else instead of two if statements so only one of the messages would be displayed to the user. Later found that the else statement would be entered if pkgs_removed is empty which would be a message like this  "The following packages were removed: " with no packages listed. Toshio said it is fine to either display only one of the messages since that was the previous way of displaying. We can change it so both are displayed.